pyiron_workflow/snippets/files.py

```diff
@@ -145,3 +145,33 @@ def _resolve_directory_and_path(
         if not isinstance(directory, DirectoryObject):
             directory = DirectoryObject(directory)
         return file_name, directory
+
+    def copy(
+        self,
+        new_file_name: str | None=None,
+        directory: DirectoryObject | str | None=None
+    ):
+        """
+        Copy an existing file to a new location.
+        Args:
+            new_file_name (str): New file name. You can also set
+                an absolute path (in which case `directory` will be ignored)
+            directory (DirectoryObject): Directory. If None, the same
+                directory is used
+        Returns:
+            (FileObject): file object of the new file
+        """
+        if new_file_name is None:
+            if directory is None:
+                raise ValueError(
```

Comment:
You could consider introducing a new error type, like 

```python
class NoDestinationError(ValueError):
    """A custom error for when neither a new file name nor new location are provided"""
```

This is a habit I'm getting into as it makes it easier in the tests to make sure that I'm actually hitting the logic branch I'm expecting to hit.